src/app_engine/client.py

```diff
@@ -2,25 +2,38 @@
 #
 # Copyright 2015 Google Inc. All Rights Reserved.
 
-"""Client model
+"""Client model.
 
 This module implements the client data model.
 """
 
-# For now we have (room_id, client_id) pairs are 'unique' but client_ids are
-# not. Uniqueness is not enforced however and bad things may happen if RNG
-# generates non-unique numbers. We also have a special loopback client id.
-# TODO(tkchin): Generate room/client IDs in a unique way while handling
-# loopback scenario correctly.
-class Client:
+import uuid
+
+
+# For OPEN type rooms, client_id is a generated random number.
+# For DIRECT type rooms, client_id is the device's gcm id.
+# (room_id, client_id) pairs are 'unique', but in the OPEN case, client_id
+# values may not be. This uniqueness is not enforced however and bad things
+# may happen if RNG generates non-unique numbers for client_ids. We also
+# have a special loopback client id.
+# The session_id value is an uuid. It is used by clients as the identifier
+# to the WSS server as well as for calling /leave in an OPEN type room.
```

Comment:
You are correct, I wrote WSS for the messaging portion for forgot the /message calls, good catch.